hibernate-gorm/src/main/groovy/io/micronaut/configuration/hibernate/gorm/HibernateDatastoreFactory.groovy

```diff
@@ -63,33 +62,20 @@ class HibernateDatastoreFactory {
             new ConfigurableEventPublisherAdapter(applicationContext),
             classes
         )
-        return datastore
-    }
-
-    @Singleton
-    SessionFactory sessionFactory(HibernateDatastore hibernateDatastore) {
-        hibernateDatastore.getSessionFactory()
-    }
-
-    @Singleton
-    DataSource dataSource(HibernateDatastore hibernateDatastore) {
-        ((HibernateConnectionSource) hibernateDatastore.getConnectionSources().defaultConnectionSource).getDataSource()
-    }
-
-    @Primary
-    @Named("hibernate")
-    @Singleton
-    PlatformTransactionManager transactionManager(HibernateDatastore hibernateDatastore) {
-        hibernateDatastore.getTransactionManager()
-    }
-
-    @Singleton
-    void setupServices(HibernateDatastore hibernateDatastore, SessionFactory sessionFactory, DataSource dataSource, PlatformTransactionManager platformTransactionManager) {
-        for (o in hibernateDatastore.getServices()) {
+        SessionFactory sessionFactory = datastore.sessionFactory
+        DataSource dataSource = ((HibernateConnectionSource) datastore.getConnectionSources().defaultConnectionSource).getDataSource()
+        PlatformTransactionManager platformTransactionManager = datastore.getTransactionManager()
+        applicationContext.registerSingleton(SessionFactory, sessionFactory)
+        applicationContext.registerSingleton(DataSource, dataSource)
+        applicationContext.registerSingleton(PlatformTransactionManager, platformTransactionManager, Qualifiers.byName("hibernate"))
+        applicationContext.registerSingleton(PlatformTransactionManager, platformTransactionManager, Qualifiers.byStereotype(Primary))
```

Comment:
I tried using a CompositeQualifier for this but the bean lookup in the test would fail no matter how I tried to do it. Registering the same bean twice with different Qualifier's seems to work fine though.